plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/AcidTables.java

```diff
@@ -133,7 +131,8 @@ public static AcidState getAcidState(TrinoFileSystem fileSystem, Location direct
 
             if (name.startsWith("base_") || name.startsWith("delta_") || name.startsWith("delete_delta_")) {
                 if (suffix.indexOf('/', slash + 1) != -1) {
-                    throw new TrinoException(HIVE_INVALID_BUCKET_FILES, "Found file in sub-directory of ACID directory: " + file.location());
+                    // Ignore nested delta directories
```

Comment:
I've been skimming https://github.com/apache/hive/blob/89005659d1d8e167208ba4f9f9aaa2de7703229d/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java (I do admit that i'm inexperienced in ORC acid), but i haven't noticed this kind of handling.
From what i read, we're taking from the **base** directory any of the `base_`, `delta_` or `delete_delta_` files and not actually listing recursively everything what's inside of the partition directory.

Is the exception we're dealing with here a consequence of the fact that we're listing recursively the partition directory?
Is there a reason why we shouldn't simply disregard the files from sub-directories ?

cc @electrum  (since you've ported the functionality from Hive)

Reply:
This seems to be skipping any ACID file in a nested directory. Any file that doesn't start with one of those names will be still treated as an "original file", which is the existing behavior. I don't remember specifically, but I assume this handling is due to us recursively listing files. We will want to verify the Hive behavior in `TestHiveAcidUtils`.

> Is there a reason why we shouldn't simply disregard the files from sub-directories ?

Not sure I understand what you're asking, as this seems to be what we do. Or do you mean that we should do this outside, and not treat them as original files?

Reply:
> Is there a reason why we shouldn't simply disregard the files from sub-directories ?

I meant initially to take into account only the files placed at the top level in the partition directory.
I think though that the change from @pajaks  covers what I meant.